pkg/vsphere/session/session.go

```diff
@@ -328,3 +345,49 @@ func isNotAuthenticated(err error) bool {
 	}
 	return false
 }
+
+// GetESXSession returns a general-purpose ESX session for tests.
+func GetESXSession(ctx context.Context, service string) (*Session, error) {
```

Comment:
Yes, I think a `test.SessionWithESX` and `test.SessionWithVPX` would be a good place for these helpers.  Don't think these helpers could be in the simulator package anyhow, since they are specific to the Session type.  Also note that Session.Populate would take care of filling in most of the Path fields as it uses the finder OrDefault methods, so they shouldn't need to be hardcoded.